tools/toolchains/arm.py

```diff
@@ -511,3 +511,21 @@ def compile(self, cc, source, object, includes):
         cmd.extend(["-o", object, source])
         cmd = self.hook.get_cmdline_compiler(cmd)
         return [cmd]
+
+    @hook_tool
+    def binary(self, resources, elf, bin):
+        _, fmt = splitext(bin)
+        # On .hex format, combine multiple .hex files (for multiple load regions) into one 
```

Comment:
This comment is no longer relevant (after copy-pase from ARMC5 impl)

Reply:
Actually, after looking at it again, why not change the ARMC5 `binary` tool?
The `ARMC6` class derives from the `ARM_STD` class which derives from the `ARM` class.